modules/core/api/router.go

```diff
@@ -0,0 +1,68 @@
+package api
+
+import (
+	"errors"
+	"fmt"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"strings"
+)
+
+// Router contains all the module-defined callbacks required by IBC Protocol V2.
+type Router struct {
+	routes map[string]IBCModule
+}
+
+// NewRouter creates a new Router instance.
+func NewRouter() *Router {
+	return &Router{
+		routes: make(map[string]IBCModule),
+	}
+}
+
+// AddRoute registers a route for a given module name.
+func (rtr *Router) AddRoute(module string, cbs IBCModule) *Router {
+	if !sdk.IsAlphaNumeric(module) {
+		panic(errors.New("route expressions can only contain alphanumeric characters"))
+	}
+
+	if rtr.HasRoute(module) {
+		panic(errors.New(fmt.Sprintf("route %s has already been registered", module)))
```

Comment:
(just curious: why do we panic with `errors.New` here rather than just `fmt.Sprintf`, as we do at line 41?)

Reply:
most panics (I think all) use errors rather than strings, so this was just to be consistent.